addons/account_payment/controllers/payment.py

```diff
@@ -42,7 +42,7 @@ def invoice_pay_form(self, acquirer_id, invoice_id, save_token=False, access_tok
         if save_token:
             vals['type'] = 'form_save'
 
-        transaction = invoice_sudo._create_payment_transaction(vals)
+        transaction = invoice_sudo.with_context(stripe_manual_payment=True)._create_payment_transaction(vals)
```

Comment:
I don't like having stripe specific context key inside a module that doesn't depend on stripe module.
I would try to put this key inside the methods that are only meant to be called for manual payments.

For example in here => https://github.com/odoo/odoo/blob/5acb2a649a2f2490725283c7f4cff5a5183d8ed0/addons/payment_stripe/models/payment.py#L56

This will create the payment transaction and return correctly the error in the case we investigated

Reply:
@tbe-odoo Indeed, I thought the same just after committing.
But I was not sure the `render` who calls the `stripe_form_generate_values` was only used in manual payment.